File: backend/src/models/post.ts
import mongoose from "mongoose";

const Schema = mongoose.Schema

let Post = new Schema({
    post_type: {
        type: String
    },
    creator_type: {
        type: Number
    },
    creator: {
        type: String
    },
    post_text: {
        type: String
    },
    post_image: {
        type: String
    },
    job_title: {
        type: String
    },
    page_name: {
        type: String
    },
    date_posted: {
        type: Number
    },
    comments: [
        {
            user: {
                type: String
            },
            profile_image: {
                type: String
            },
            text: {
                type: String
            }
        }
    ]
})

export default mongoose.model('Post', Post, 'post')

File: WorkHub/app/src/main/java/com/example/workhub/data/retrofit/models/Post.kt
package com.example.workhub.data.retrofit.models

data class Post(
    val _id: String,
    val post_type: String,
    val creator_type: Int,
    val creator: String,
    val post_text: String,
    val post_image: String,
    val job_title: String,
    val page_name: String,
    val date_posted: Long,
    var comments: List<Comment>
)

data class Comment(
    val user: String,
    val profile_image: String,
    val text: String
)
